proposed/local-sdk-global-json.md

```diff
@@ -0,0 +1,215 @@
+# Provide SDK hint paths in global.json
+
+## Summary
+
+This proposal adds two new properties to the `sdk` object in
+[global.json][global-json-schema]
+
+```json
+{
+    "sdk": {
+        "paths": [ ".dotnet", "$host" ],
+        "errorMessage": "The .NET SDK could not be found, please run ./install.sh."
+    }
+}
+```
+
+These properties will be considered by the resolver host during .NET SDK
+resolution. The `paths` property lists the locations that the resolver should
+consider when attempting to locate a compatible .NET SDK. The `errorMessage`
+property controls what the resolver displays when it cannot find a compatible
+.NET SDK.
+
+This particular configuration would cause the local directory `.dotnet` to be
+considered _in addition_ to the current set of locations. Further if resolution
+failed the resolver would display the contents of `errorMessage` instead of
+the default error message.
+
+## Motivation
+
+There is currently a disconnect between the ways the .NET SDK is deployed in
+practice and what the host resolver can discover when searching for compatible
+SDKs. By default the host resolver is only going to search for SDKs in machine
+wide locations: `C:\Program Files\dotnet`, `%PATH%`, etc ...  The .NET SDK
```

Comment:
```suggestion
SDKs. By default the host resolver is only going to search for SDKs next to
the running `dotnet`. This often means machine-wide locations, since users
and tools typically rely on `dotnet` already being on the user's path when
launching, instead of specifying a full path to the executable. The .NET SDK
```